### PR TITLE
Code rev

### PR DESCRIPTION
**Pull request by MostafaShahat2021**
Hi @zeff96, 👋, Hope all is well with you, 
please review my code 💯 👍 
### In this project we :

- Add a function for adding a new task.
- Add a function for deleting a task.
- Add a function for editing task descriptions.
- Add function for Updating items object's value for completed key upon user actions.
- Add function for clearing all button
- Use webpack to bundle JavaScript.
- Use NPM.
- Use webpack.
- Use webpack-cli.
- Use css-loader.
- Use style-loader.
- Modify webpack.config.js
- use HtmlWebpackPlugin.
- Setup local dev server
- Add Linters for HTML/CSS.
- Use best practices for HTML & CSS.
- Update README file.
- Add Mark up to the .html file.
- Add style to the website.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+/* eslint-disable import/no-mutable-exports */
+import './style.css';
+
+const submit = document.querySelector('.add-btn');
+const tasksDiv = document.querySelector('.tasks');
+export const taskInput = document.querySelector('.add-task input');
+
+let editId;
+let isEditTask = false;
+let tasks = JSON.parse(localStorage.getItem('task-list'));
```
Comment (zeff96) [inline src/index.js:11]: I would recommend reducing the number of global variables in a given project. And if need be of using more than one global variable then I would recommend closures. What do you think?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+export const addElemToPage = () => {
```
Comment (zeff96) [inline src/index.js]: I would advise you to look out for unused functions, They make your code a bit messy. what do you think?

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  let div = '';
+  if (tasks) {
+    tasks.forEach((task) => {
+      let isCompleted;
+      if (task.completed === true) {
+        isCompleted = 'checked';
+      } else {
+        isCompleted = '';
+      }
```
Comment (zeff96) [inline src/index.js:26]: What do you think of getting the id of the checkbox and comparing it to the index of the task to make your code look shorter?

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      tasks.push(task);
+    } else {
+      isEditTask = false;
+      tasks[editId].descripton = userTask;
+    }
+    taskInput.value = '';
+    localStorage.setItem('task-list', JSON.stringify(tasks));
+    addElemToPage();
+  }
+});
```
Comment (zeff96) [inline src/index.js]: I would suggest to you use a function to generate this part to help keep your CODE dry, or if you must use evenlistener then I would recommend you add to to the parent container with containing both the input and button

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  if (selectedTask.checked) {
+    taskName.classList.add('checked');
+    tasks[selectedTask.id - 1].completed = true;
+  } else {
+    taskName.classList.remove('checked');
+    tasks[selectedTask.id - 1].completed = false;
+  }
+  localStorage.setItem('task-list', JSON.stringify(tasks));
+};
+
```
Comment (zeff96) [inline src/index.js]: Don't you think it needs to be more clean and dry, I know you can do better.

File: src/index.html [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            <input type="text" name="" id="task" placeholder="Add to your list">
+            <button class="add-btn"> <span class="material-symbols-outlined">
+                    keyboard_return
+                </span></button>
+            </div>
+
+        <div class="tasks"></div>
+        <p class="clear"><a href="#">clear all completed</a></p>
+    </main>
+</body>
```
Comment (zeff96) [inline src/index.html:33]: Kindly check on the spaces indentation and make sure it is set to 2 space

**Review by zeff96 — CHANGES_REQUESTED** [review]
Hi @MostafaShahat2021 ,
Good job so far👋
There are some changes that are required before you move on to the next project but you are almost there!
Alt Text
## Highlights
✅  Good Gitflow,
✅ No linter errors,
✅ Good commits

## Required Changes♻️
Check the comments under the review

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

File: src/index.js (hunk 2 above)
Comment (MostafaShahat2021) [reply src/index.js]: thanks, I'll fix that 👍

File: src/index.html (hunk 6 above)
Comment (MostafaShahat2021) [reply src/index.html:33]: thanks, I'll fix that 👍

**Comment by MostafaShahat2021** [comment]
Hi @zeff96 👋, hope all is well with you, Thanks a lot for your efforts in reviewing my code 💯.
actually, I totally agree with you about all the changes you asked me to do, but it needs to refactor the whole code,
so I'll consider your required changes and I'll apply them in the near future, thanks again for all your efforts 😄.

**Review by zeff96 — APPROVED** [review]
@MostafaShahat2021 ,
Thanks for your feedback, your project is now approved, nothing else to say rather than it's time to merge it 💯

**Comment by MostafaShahat2021** [comment]
Thanks a million 💯 👍 .

**Pull request merged**